tests/integration/callbacks/test_callback_context.py

```diff
@@ -96,3 +99,247 @@ def report_triggered(n):
         'triggered is truthy, has prop/id ["btn", "n_clicks"], and full value '
         '[{"prop_id": "btn.n_clicks", "value": 1}]',
     )
+
+
+calls = 0
+callback_contexts = []
+clicks = dict()
+
+
+@pytest.mark.DASH1350
+def test_cbcx005_grouped_clicks(dash_duo):
+    app = Dash(__name__)
+    app.layout = html.Div(
+        [
+            html.Button("Button 0", id="btn0"),
+            html.Div(
+                [
+                    html.Button("Button 1", id="btn1"),
+                    html.Div(
+                        [html.Div(id="div3"), html.Button("Button 2", id="btn2")],
+                        id="div2",
+                        style=dict(backgroundColor="yellow", padding="50px"),
+                    ),
+                ],
+                id="div1",
+                style=dict(backgroundColor="blue", padding="50px"),
+            ),
+        ],
+        id="div0",
+        style=dict(backgroundColor="red", padding="50px"),
+    )
+
+    @app.callback(
+        Output("div3", "children"),
+        [
+            Input("div1", "n_clicks"),
+            Input("div2", "n_clicks"),
+            Input("btn0", "n_clicks"),
+            Input("btn1", "n_clicks"),
+            Input("btn2", "n_clicks"),
+        ],
+        prevent_initial_call=True,
+    )
+    def update(div1, div2, btn0, btn1, btn2):
+        global calls
+        global callback_contexts
+        global clicks
```

Comment:
Rather than globals, most of our tests use `multiprocessing.Value` for this type of situation - a little cleaner I think, and doesn't depend on threading vs processes when the server and selenium parts branch.

https://github.com/plotly/dash/blob/92684802c94f0ff408c7747e3e8e650aaab43850/tests/integration/callbacks/test_multiple_callbacks.py#L18

Reply:
Since `callback_contexts` is an array of dicts and `clicks` is a dict, I don't think I can easily use `Value` as it expects simple types. Instead used a scoped `context` class that should not leak outside of the test function. https://github.com/plotly/dash/pull/1385/commits/90697b15c5600587b36c0e67ffc0a6fe89b03b4a

Reply:
Well ok, I won't insist - it's probably fine, since we should always be using threads. I still like the `multiprocessing` pattern better as a general rule though, just to make it crystal clear that you're using this to communicate between the server thread and the selenium thread.